bfm/tests/CMakeLists.txt

```diff
@@ -15,25 +15,48 @@
 # You should have received a copy of the GNU Lesser General Public
 # License along with this library; if not, write to the Free Software
 # Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+cmake_minimum_required(VERSION 3.6)
+project(bfm_test C CXX)
+include(${BF_SCRIPTS_DIR}/cmake/flags/flags.cmake)
+include(CTest)
+enable_testing(true)
+enable_language(ASM_NASM)
 
-include(${CMAKE_INSTALL_PREFIX}/cmake/CMakeGlobal_Includes.txt)
+set(PROJECT_SRC_DIR ${CMAKE_CURRENT_LIST_DIR}/../src)
+set(PROJECT_TEST_DIR ${CMAKE_CURRENT_LIST_DIR}/../tests)
+set(PROJECT_INC_DIR ${CMAKE_CURRENT_LIST_DIR}/../include)
 
-# ------------------------------------------------------------------------------
-# Targets
-# ------------------------------------------------------------------------------
+include_directories(
+    ${PROJECT_INC_DIR}
+    ${BUILD_SYSROOT_TEST}/include
+    ${BUILD_SYSROOT_OS}/include
+)
 
-add_library(test_vmcall STATIC test_vmcall.cpp)
+link_directories(
+    ${BUILD_SYSROOT_TEST}/lib
+)
+
+set(BFM_TEST_CXX_FLAGS ${CMAKE_CXX_FLAGS} "-DENABLE_UNITTESTING=1")
+generate_flags(USERSPACE ADD_CXX_FLAGS ${BFM_TEST_CXX_FLAGS})
+
+add_subdirectory(${PROJECT_SRC_DIR}/../ioctl ${CMAKE_CURRENT_BINARY_DIR}/ioctl-src)
+add_subdirectory(${PROJECT_SRC_DIR} ${CMAKE_CURRENT_BINARY_DIR}/src)
+add_subdirectory(${PROJECT_SRC_DIR}/../main ${CMAKE_CURRENT_BINARY_DIR}/main-src)
+
+add_library(test_bfm_support STATIC test_vmcall.cpp)
+install(TARGETS test_bfm_support DESTINATION ${BUILD_SYSROOT_TEST}/lib)
```

Comment:
Samething here. Why are we installing support libraries? Please clean this up for all instances.